metabase/metabase
# Fix oom during sync field values

Fixes https://github.com/metabase/metabase/issues/46411 by reworking the `field-distinct-values` function so that it stops getting rows from jdbc if the total char of all rows exceeds the char limit.


### How to test:
1. Create a db with this script

```sql
CREATE TABLE example_table (id SERIAL PRIMARY KEY,
                            content TEXT);

INSERT INTO example_table (content)
SELECT
    substr(
        -- Generate a long random string
        repeat(
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text) ||
            md5(random()::text),
            90909  -- Number of repetitions to exceed 1,000,000 characters
        ),
        1,
        200000
    )
FROM generate_series(1, 999);
```
2. Start MB with a max heap size of 512mb
3. Sync the DB should succeed
